vllm/v1/spec_decode/eagle.py

```diff
@@ -302,12 +303,25 @@ def load_model(self, target_model: nn.Module) -> None:
         self.attn_layer_name = next(iter(draft_attn_layer_names))
         loaded_weights = self.model.load_weights(
             loader.get_all_weights(draft_model_config, self.model))
-        if self.vllm_config.speculative_config.method == "eagle3":
-            if "model.embed_tokens.weight" not in loaded_weights:
-                logger.info(
-                    "Loading EAGLE embedding weights from the target model.")
-                self.model.model.embed_tokens = target_model.model.embed_tokens
+
+        # share embed_tokens with the target model if needed
+        if get_pp_group().world_size == 1:
+            assert "model.embed_tokens.weight" not in loaded_weights, \
+            "For PP = 1, Eagle draft should share embed with target model"
+            logger.info(
+                "Loading EAGLE embedding weights from the target model.")
+            self.model.model.embed_tokens = target_model.model.embed_tokens
         else:
+            assert "model.embed_tokens.weight" in loaded_weights, \
+            "For PP > 0, Eagle draft checkpoint should its own copy of "
+            " the model.embed_tokens.weight"
+            logger.info("EAGLE embedding weights are already loaded.")
```

Comment:
```suggestion
            logger.info("Since PP > 1, the EAGLE head loaded its own vocab embedding weights instead of sharing them with the target model.")
```